front/lib/iam/users.ts

```diff
@@ -95,46 +89,56 @@ export async function maybeUpdateFromExternalUser(
 
 export async function createOrUpdateUser(
   session: SessionWithUser
-): Promise<{ user: UserType; created: boolean }> {
+): Promise<{ user: UserResource; created: boolean }> {
   const { user: externalUser } = session;
 
   const user = await fetchUserFromSession(session);
 
   if (user) {
+    const updateArgs = {
+      username: user.username,
+      firstName: user.firstName,
+      lastName: user.lastName,
+      email: user.email,
+    };
     // We only update the user's email if the email is verified.
     if (externalUser.email_verified) {
-      user.email = externalUser.email;
+      updateArgs.email = externalUser.email;
     }
 
     // Update the user object from the updated session information.
-    user.username = externalUser.nickname;
-    user.name = externalUser.name;
+    updateArgs.username = externalUser.nickname;
 
     if (!user.firstName && !user.lastName) {
       if (externalUser.given_name && externalUser.family_name) {
-        user.firstName = externalUser.given_name;
-        user.lastName = externalUser.family_name;
+        updateArgs.firstName = externalUser.given_name;
+        updateArgs.lastName = externalUser.family_name;
       } else {
         const { firstName, lastName } = guessFirstAndLastNameFromFullName(
           externalUser.name
         );
-        user.firstName = firstName;
-        user.lastName = lastName;
+        updateArgs.firstName = firstName;
+        updateArgs.lastName = lastName;
       }
     }
 
-    await user.save();
+    await user.updateInfo(
```

Comment:
This is different from what we are currently doing, as `user.save()` was not doing a "saving" if the user has not been modified.